frontend/public/components/utils/webhooks.tsx

```diff
@@ -56,17 +56,31 @@ export const WebhookTriggers: React.FC<WebhookTriggersProps> = (props) => {
   });
   const tableColumnClasses = getTableColumnClasses(canGetSecret);
   const [webhookSecrets, setWebhookSecrets] = React.useState<K8sResourceKind[]>([]);
+  const [webhookTriggers, setWebhookTriggers] = React.useState<WebhookTrigger[]>([]);
+  const [secretNames, setSecretNames] = React.useState<string[]>([]);
   const [secretErrors, setSecretErrors] = React.useState<string[]>([]);
   const [isLoaded, setLoaded] = React.useState(false);
 
-  const webhooks: WebhookTrigger[] = _.filter(triggers, ({ type }) => webhookTriggers.has(type));
-  const secretNames: string[] = _.uniq(
-    webhooks.reduce((acc: string[], webhook: WebhookTrigger): string[] => {
-      const triggerProperty = getTriggerProperty(webhook);
-      const secretName = _.get(webhook, [triggerProperty, 'secretReference', 'name']);
-      return secretName ? [...acc, secretName] : acc;
-    }, []),
-  );
+  React.useEffect(() => {
+    const newWebhookTriggers: WebhookTrigger[] = _.filter(triggers, ({ type }) =>
+      webhookTriggerTypes.has(type),
+    );
+    if (_.isEqual(newWebhookTriggers, webhookTriggers)) {
```

Comment:
Recommend using functional form of update state here, that way we don't have to have webHookTriggers as a dep in this effect. https://reactjs.org/docs/hooks-reference.html#functional-updates

```tsx
  React.useEffect(() => {
    setWebhookTriggers((previousTriggers) => {
      const newTriggers = _.filter(triggers, ({ type }) => webhookTriggerTypes.has(type));
      return _.isEqual(newTriggers, previousTriggers) ? previousTriggers : newTriggers;
    });
  }, [triggers]);
```